modules/data-streams/src/test/java/org/elasticsearch/datastreams/mapper/DataStreamTimestampFieldMapperTests.java

```diff
@@ -231,4 +234,283 @@ public void testValidateDefaultIgnoreMalformed() throws Exception {
             assertThat(summaryTimestamp.ignoreMalformed(), is(false));
         }
     }
+
+    public void testFieldTypeWithDocValuesSkipper_LogsDBModeDisabledDocValuesSkipper() throws IOException {
+        final Settings settings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
+            .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), DataStreamTimestampFieldMapper.DEFAULT_PATH)
+            .put(IndexSettings.USE_DOC_VALUES_SKIPPER.getKey(), false)
+            .build();
+        final MapperService mapperService = createMapperService(settings, timestampMapping(true, b -> {
+            b.startObject(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+            b.field("type", "date");
+            b.endObject();
+        }));
+
+        final DateFieldMapper timestampMapper = (DateFieldMapper) mapperService.documentMapper()
+            .mappers()
+            .getMapper(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+        assertTrue(timestampMapper.fieldType().hasDocValues());
+        assertFalse(timestampMapper.fieldType().hasDocValuesSkipper());
+        assertTrue(timestampMapper.fieldType().isIndexed());
+    }
+
+    public void testFieldTypeWithDocValuesSkipper_LogsDBMode() throws IOException {
+        final Settings settings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
+            .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), DataStreamTimestampFieldMapper.DEFAULT_PATH)
+            .build();
+        final MapperService mapperService = createMapperService(settings, timestampMapping(true, b -> {
+            b.startObject(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+            b.field("type", "date");
+            b.endObject();
+        }));
+
+        final DateFieldMapper timestampMapper = (DateFieldMapper) mapperService.documentMapper()
+            .mappers()
+            .getMapper(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+        assertTrue(timestampMapper.fieldType().hasDocValues());
+        if (IndexSettings.USE_DOC_VALUES_SKIPPER.get(settings)) {
+            assertFalse(timestampMapper.fieldType().isIndexed());
+            assertTrue(timestampMapper.fieldType().hasDocValuesSkipper());
+        } else {
+            assertTrue(timestampMapper.fieldType().isIndexed());
+            assertFalse(timestampMapper.fieldType().hasDocValuesSkipper());
+        }
```

Comment:
Let's just add an assumption with feature flag here? Otherwise when we remove the feature flag, this else statement will never be invoked and I think future readers of the this test may wonder how the index setting gets set?

Reply:
`assumeTrue(IndexSettings.DOC_VALUES_SKIPPER.isEnabled())`